### ISSUE TITLE
telegraf_plugins_extra is override when multiple call

### ISSUE THREAD

**Issue by sinc59** [dj-wasabi/ansible-telegraf#30]
Hello,
i use this role with multiple groups (postgres_database, rabbitmq)
For this, i use groups_vars files to specify extra_plugins (group_vars/postgres_database, group_vars/rabbitmq)
These files contains:

- group_vars/postgres_database:

```
telegraf_plugins_extra:
    - plugin: postgresql
      config:
          - address = "postgres://user:password@localhost/postgres"
```

- group_vars/rabbitmq
```
telegraf_plugins_extra:
    - plugin: rabbitmq
      config:
        - url = "http://localhost:15672"
```
 
If a server is one two roles (database and rabbitmq), the file /etc/telegraf/telegraf.d/extra-plugins.conf contain only the rabbitmq.

Do you know how i can aggregate this two have the postgresql and rabbitmq in this server ?

Best regards

**Comment by NikonNLG** [comment]
@sinc59 did you take a look at http://docs.ansible.com/ansible/latest/intro_configuration.html#hash-behaviour ?
you can try `hash_behaviour = merge`, but be care to not merge other settings.

**Comment by alekseyp** [comment]
Same issue.

**Comment by bengoa** [comment]
I think you could use `filename` directive to generate different file names:

- group_vars/postgres_database:
```
telegraf_plugins_extra:
    - plugin: postgresql
      filename: postgresql
      config:
          - address = "postgres://user:password@localhost/postgres"
```

- group_vars/rabbitmq
```
telegraf_plugins_extra:
    - plugin: rabbitmq
      filename: rabbitmq
      config:
        - url = "http://localhost:15672"
```

**Comment by tjend** [comment]
This was solved in 0.9.0. See https://github.com/dj-wasabi/ansible-telegraf/pull/46.

You will need to change your config to a hash instead of a list, and need hash_behaviour=merge in your ansible.cfg.

- group_vars/postgres_database:

```
telegraf_plugins_extra:
  postgresql:
    filename: postgresql
    config:
      - address = "postgres://user:password@localhost/postgres"
```

- group_vars/rabbitmq
```
telegraf_plugins_extra:
  rabbitmq:
    filename: rabbitmq
    config:
      - url = "http://localhost:15672"
```

**Comment by dj-wasabi** [comment]
Closing issue as fix is applied in a PR.
